### PR TITLE
Simplified mount interface

### PR DESCRIPTION
**Pull request by rafalcieslak**
Initially I only wanted to remove the `if (strncmp(path, "/dev/", 5) == 0)` and associated `vfs_root_dev_vnode`, turned out some extra cleanup to the mount process was required, and I ended up with an elegant
```c
vfs_domount_named("initrd", "/");
vfs_domount_named("devfs", "/dev");
```
in `kernel_init(...)`.

### CODE REVIEW

**Comment by cahirwpz** [comment]
`vfs_domount_named` looks to me like a good candidate for a function that would implement [mount](http://netbsd.gw.com/cgi-bin/man-cgi?mount+2+NetBSD-current) syscall! We wouldn't use last three arguments yet, but why not introduce it right now?

**Comment by rafalcieslak** [comment]
I've considered that, but I don't think a `mount` syscall would be of much value. I suppose we wouldn't use it until we have some reasonable init program.

**Comment by cahirwpz** [comment]
That's right, but we could make some provisions for the future, e.g. by renaming `vfs_mount_named` to `do_mount`, at least.

File: sys/vfs_syscalls.c [hunk 1]
```diff
@@ -173,3 +186,27 @@ int sys_fstat(thread_t *td, syscall_args_t *args) {
     return error;
   return 0;
 }
+
+int sys_mount(thread_t *td, syscall_args_t *args) {
+  char *user_fsysname = (char *)args->args[0];
+  char *user_pathname = (char *)args->args[1];
+
+  int error = 0;
+  char fsysname[256];
```
Comment (cahirwpz) [inline sys/vfs_syscalls.c]: We must stop using kernel stack so carelessly... after all we have only a page (or two?) of it! Please look for reference at FreeBSD [sys_mount](http://bxr.su/FreeBSD/sys/kern/vfs_mount.c#731) – the actually use `M_TEMP` memory pool to `copyinstr`.

File: include/mount.h [hunk 2]
```diff
@@ -42,6 +42,11 @@ typedef struct vfsconf {
   TAILQ_ENTRY(vfsconf) vfc_list; /* Entry on the list of vfsconfs */
 } vfsconf_t;
 
+/* The list of all installed filesystem types */
+typedef TAILQ_HEAD(, vfsconf) vfsconf_list_t;
+extern vfsconf_list_t vfsconf_list;
```
Comment (cahirwpz) [inline include/mount.h]: Wouldn't it make more sense for now to have linker set to store those?

Comment (rafalcieslak) [inline include/mount.h]: Huh? We do use a linker set for kernel-provided filesystems. Please look at `vfs_init()`.

Comment (cahirwpz) [inline include/mount.h]: Oops. I guess I'm not that familiar with this part of the kernel.

File: sys/vfs_syscalls.c [hunk 3]
```diff
@@ -173,3 +187,32 @@ int sys_fstat(thread_t *td, syscall_args_t *args) {
...
+  error = copyinstr(user_pathname, pathname, sizeof(pathname), &n);
+  if (error < 0)
+    goto end;
+
+  log("mount(\"%s\", \"%s\")", pathname, fsysname);
+
+  error = do_mount(td, fsysname, pathname);
+end:
+  kfree(M_TEMP, fsysname);
+  kfree(M_TEMP, pathname);
```
Comment (rafalcieslak) [inline sys/vfs_syscalls.c]: BTW, @coodie `scoped_ptr` here?

File: sys/malloc.c [hunk 4]
```diff
@@ -267,3 +267,5 @@ void kmem_dump(kmem_pool_t *mp) {
 /* TODO: missing implementation */
 void kmem_destroy(kmem_pool_t *mp) {
 }
+
+MALLOC_DEFINE(M_TEMP, "temporaries pool", 2, 4);
```
Comment (coodie) [inline sys/malloc.c]: I like idea behind this pool. Going to be easier to implement some tests in other branches where I needed to perform some memory allocation. It wouldn't hurt to increase it's size to be even bigger (say 32 pages).

File: sys/vfs_syscalls.c [hunk 5]
```diff
@@ -85,6 +88,17 @@ int do_fstat(thread_t *td, int fd, vattr_t *buf) {
   return res;
 }
 
+int do_mount(thread_t *td, const char *fs, const char *path) {
```
Comment (coodie) [inline sys/vfs_syscalls.c]: We would want to extend interface of mount to have extra flags argument, so we can have 'read only', 'no exec' and possibly other types of mounts. Not a big deal and can be added in future.

File: sys/vfs_syscalls.c (hunk 3 above)
Comment (coodie) [inline sys/vfs_syscalls.c]: Yes, scoped_ptr would be great in this case :), but it's not part of this PR.
